lib/blockchain_double.js

```diff
@@ -399,7 +411,7 @@ BlockchainDouble.prototype.getQueuedNonce = function(address, callback) {
   if (nonce != null) {
     return callback(null, Buffer.from([nonce + 1]));
   }
-  this.vm.stateManager.getAccount(addressBuffer, function(err, val) {
+  this.stateTrie.get(addressBuffer, function(err, val) {
```

Comment:
I had recently changed this line to `this.vm.stateManager.getAccount` but that is actually wrong as we _shouldn't_ be reading from the cache here, as the cache could be dirty (I don't think i can be dirty now, but in the future when we enable `asyncRequestProcessing` I think it could be). We should actually always be going to the trie itself when reading out data like this.